chunk.go

```diff
@@ -86,8 +86,12 @@ func (node *Node) traverseDepth(t *Tree, depth uint, cb func(*Node)) {
 	}
 
 	// otherwise, decend one more level
-	node.getLeftNode(t).traverseDepth(t, depth-1, cb)
-	node.getRightNode(t).traverseDepth(t, depth-1, cb)
+	if node.leftHash != nil {
```

Comment:
This will not work for trees that have some unsaved state (if leftHash is nil, but leftNode is not nil, then it is valid, just not saved).

That said, we only try to make chunks for persisted trees, so this is fine. Just comment please.

Reply:
Good point, maybe I can add a check too.

Reply:
Actually can't we just check if it's not a leaf? looks like tests pass.